core/src/main/scala/cats/data/OneAnd.scala

```diff
@@ -12,108 +13,125 @@ package data
 
+  /**
+   * Typesafe equality operator.
+   *
+   * This method is similar to == except that it only allows two
+   * OneAnd[A, F] values to be compared to each other, and uses
+   * equality provided by Eq[_] instances, rather than using the
+   * universal equality provided by .equals.
+   */
   def ===(that: OneAnd[A, F])(implicit A: Eq[A], FA: Eq[F[A]]): Boolean =
     A.eqv(head, that.head) && FA.eqv(tail, that.tail)
 
+  /**
+   * Typesafe stringification method.
+   *
+   * This method is similar to .toString except that it stringifies
+   * values according to Show[_] instances, rather than using the
+   * universal .toString method.
+   */
   def show(implicit A: Show[A], FA: Show[F[A]]): String =
     s"OneAnd(${A.show(head)}, ${FA.show(tail)})"
 }
 
 trait OneAndInstances {
 
-  implicit def oneAndEq[A, F[_]](implicit A: Eq[A], FA: Eq[F[A]]): Eq[OneAnd[A, F]] = new Eq[OneAnd[A, F]]{
-    def eqv(x: OneAnd[A, F], y: OneAnd[A, F]): Boolean = x === y
-  }
+  implicit def oneAndEq[A, F[_]](implicit A: Eq[A], FA: Eq[F[A]]): Eq[OneAnd[A, F]] =
+    new Eq[OneAnd[A, F]]{
+      def eqv(x: OneAnd[A, F], y: OneAnd[A, F]): Boolean = x === y
+    }
 
   implicit def oneAndShow[A, F[_]](implicit A: Show[A], FA: Show[F[A]]): Show[OneAnd[A, F]] =
     Show.show[OneAnd[A, F]](_.show)
 
-  implicit def oneAndFunctor[F[_]](F: Functor[F]): Functor[OneAnd[?,F]] = new Functor[OneAnd[?,F]] {
-    override def map[A, B](fa: OneAnd[A,F])(f: A => B): OneAnd[B, F] =
-      OneAnd(f(fa.head), F.map(fa.tail)(f))
-  }
+  implicit def oneAndFunctor[F[_]](F: Functor[F]): Functor[OneAnd[?, F]] =
+    new Functor[OneAnd[?, F]] {
+      def map[A, B](fa: OneAnd[A, F])(f: A => B): OneAnd[B, F] =
+        OneAnd(f(fa.head), F.map(fa.tail)(f))
+    }
 
-  implicit def oneAndSemigroupK[F[_] : MonadCombine]: SemigroupK[OneAnd[?,F]] = new SemigroupK[OneAnd[?,F]] {
-    def combine[A](a: OneAnd[A, F], b: OneAnd[A, F]): OneAnd[A, F] = a combine b
-  }
+  implicit def oneAndSemigroupK[F[_]: MonadCombine]: SemigroupK[OneAnd[?, F]] =
+    new SemigroupK[OneAnd[?, F]] {
+      def combine[A](a: OneAnd[A, F], b: OneAnd[A, F]): OneAnd[A, F] =
+        a combine b
+    }
 
-  implicit def oneAndFoldable[F[_]](implicit foldable: Foldable[F]): Foldable[OneAnd[?,F]] = new Foldable[OneAnd[?,F]] {
-    override def foldLeft[A,B](fa: OneAnd[A,F], b: B)(f: (B,A) => B): B =
-      fa.foldLeft(b)(f)
-    override def foldRight[A,B](fa: OneAnd[A,F], b: B)(f: (A,B) => B): B =
-      fa.foldRight(b)(f)
-
-    override def partialFold[A, B](fa: OneAnd[A,F])(f: A => Fold[B]): Fold[B] = {
-      import Fold._
-      f(fa.head) match {
-        case b @ Return(_) => b
-        case Continue(c) => foldable.partialFold(fa.tail)(f) match {
-          case Return(b) => Return(c(b))
-          case Continue(cc) => Continue { b => c(cc(b)) }
-          case _ => Continue(c)
+  implicit def oneAndFoldable[F[_]](implicit foldable: Foldable[F]): Foldable[OneAnd[?,F]] =
```

Comment:
I think it should wait for another PR as this one is already pretty large, but it looks like we are missing a `oneAndReducible` instance. Is that right?